sameproject/program/compile/notebook_processing.py

```diff
@@ -37,6 +37,18 @@ def read_notebook(notebook_path) -> dict:
     return ntbk_dict
 
 
+def parse_magic_lines(code: str) -> List[str]:
+    parser = jupytext.magics.StringParser("python")
+
+    magic_lines = []
+    for i, line in enumerate(code.split("\n")):
+        if not parser.is_quoted() and jupytext.magics.is_magic(line, "python"):
```

Comment:
Other changes are because I branched off another PR branch - this is the main change. We can use `jupytext.magics.StringParser` to check whether a line is in the context of a comment or quote.